language/move-binary-format/src/check_bounds.rs

```diff
@@ -123,9 +182,14 @@ impl<'a> BoundsChecker<'a> {
     }
 
     fn check_field_handle(&self, field_handle: &FieldHandle) -> PartialVMResult<()> {
-        check_bounds_impl(&self.module.struct_defs, field_handle.owner)?;
+        check_bounds_impl(&self.view.struct_defs().unwrap_or(&[]), field_handle.owner)?;
         // field offset must be in bounds, struct def just checked above must exist
-        if let Some(struct_def) = &self.module.struct_defs.get(field_handle.owner.into_index()) {
+        if let Some(struct_def) = &self
+            .view
+            .struct_defs()
+            .unwrap_or(&[])
+            .get(field_handle.owner.into_index())
```

Comment:
maybe add `struct_def_at` and have it return an option

Reply:
Yeah, I think I considered adding or using `*_at` methods in this file, but I decided against it. Looking at the other `BinaryIndexedView::*_at` methods, they all use index expressions `[]` to access the elements directly, and panic if the index is out-of-bounds. That struck me as pretty different from the `get` methods used in this file, which return `Option`.

My take here would be: if we want to clean up some of these expressions by implementing something like `struct_def_at` or (elsewhere in the file) `struct_handle_at`, etc., I would name them differently, to make a distinction between panicking and non-panicking methods. Maybe something like `get_struct_def` or something? What do you think?

Reply:
maybe all of this is fine? I don't know... I think in general the unwrap_or is a bit weird to me and might prefer `.struct_defs().then(|defs| defs.field_handle.owner.into_index())`

Reply:
In general, all of the indexed stuff uses `struct_def_at` because it it was guaranteed to be over a bounds-checked module (which isn't the case anymore)